src/ui/public/style_compile/style_compile.css.tmpl

```diff
@@ -5,3 +5,21 @@
 .truncate-by-height:before {
   top: <%= truncateGradientTop %>;
 }
+
+<% if (disableAnimations) { %>
+*:not(.essentialAnimation),
+*:not(.essentialAnimation):before,
+*:not(.essentialAnimation):after {
+
+  /**
+   * set the animation/transition duration to 0s so that animation callbacks are
+   * still triggered, allowing components that require them to remain functional
+   */
+
+  -webkit-animation-duration: 0s !important;
+  animation-duration: 0s !important;
+
+  -webkit-transition-duration: 0s !important;
+  transition-duration: 0s !important;
+}
+<% } %>
```

Comment:
Since we're not injecting any JS variables here, would it make more sense to use a system similar to how we apply EUI themes?

Reply:
Here's a reference: https://github.com/elastic/kibana/pull/15657/files#diff-8483334633f95ed911cae02b98f54f02R15

Reply:
Yeah, I considered something like that, but none of the existing style-inclusion features really seemed to fit in my eyes, but this one felt the most similar because it's dynamically defining the css in a style block based on config.

Maybe I could add a `theme.extendTheme(css)` method that will include additional css in the theme style element every time, regardless what the theme is, or maybe I could use the the ["usable" feature of the webpack `style-loader`](https://github.com/webpack-contrib/style-loader#useable) to dynamically add and remove just a single style sheet, or maybe we just want a module in `ui/styles` somewhere that watches the config and adds/removes an element from the `<head>` based on the config... Which would you choose?